akka-stream/src/main/scala/akka/stream/javadsl/Source.scala

```diff
@@ -628,6 +628,30 @@ final class Source[+Out, +Mat](delegate: scaladsl.Source[Out, Mat]) extends Grap
     new Source(delegate.prependMat(that)(combinerToScala(matF)))
 
   /**
+   * Provides an alternative that will be consumed if this source completes without any
+   * elements passing by. As soon as the first element comes through this stream, the alternative
+   * will be cancelled.
```

Comment:
I think I projected my usual worry from concat onto this element (do we pull the 2nd one early or not - even if not signalling downstream). It kind of is explained that we don't by the name of the element, so perhps ignore this commetn